Wi-Fi: Compilation Errors

**Description**
The changes made to the Wi-fi shell it fails to compile(Add support to configure BSSID in Wi-Fi connect command #70024). This is a known issue with C case statements.

> /subsys/net/l2/wifi/wifi_shell.c:507:4: error: a label can only be part of a statement and a declaration is not a statement
>   507 |    long channel = strtol(optarg, &endptr, 10);

**Impact**
Showstopper

**Logs and console output**

> /subsys/net/l2/wifi/wifi_shell.c:507:4: error: a label can only be part of a statement and a declaration is not a statement
>   507 |    long channel = strtol(optarg, &endptr, 10);




**Additional context**
By adding an additional statement the compiler ignores that 'issue'
```
		case 'c': //Break Comment to resolve compile error
			long channel = strtol(optarg, &endptr, 10);
```

--- a/subsys/net/l2/wifi/wifi_shell.c
+++ b/subsys/net/l2/wifi/wifi_shell.c
@@ -470,6 +470,7 @@ static int __wifi_args_to_params(const struct shell *sh, size_t argc, char *argv
 	bool found = false;
 	char bands_str[MAX_BANDS_STR_LEN] = {0};
 	size_t offset = 0;
+	long channel;
 
 	/* Defaults */
 	params->band = WIFI_FREQ_BAND_UNKNOWN;
@@ -504,7 +505,7 @@ static int __wifi_args_to_params(const struct shell *sh, size_t argc, char *argv
 			}
 			break;
 		case 'c':
-			long channel = strtol(optarg, &endptr, 10);
+			channel = strtol(optarg, &endptr, 10);
 			for (band = 0; band < ARRAY_SIZE(all_bands); band++) {
 				offset += snprintf(bands_str + offset,
 						   sizeof(bands_str) - offset,

